tests/internal/test_settings.py

```diff
@@ -605,3 +605,24 @@ def test_remoteconfig_header_tags(run_python_code_in_subprocess):
         env=env,
     )
     assert status == 0, f"err={err.decode('utf-8')} out={out.decode('utf-8')}"
+
+
+def _stderr_contains_log(stderr: str) -> bool:
+    return stderr.startswith("Finished span not connected to a trace, adding to trace.")
```

Comment:
I think tests on stderr are finicky and end up making life difficult if one of us does something that changes logging (from an unrelated part of the tracer) later on.

IMO, a test on the logger that checks that it was called with the expected args is less flaky and more maintainable.

Reply:
could always just make this a `"Finished span not connected ..." in stderr` too.

I'd love to keep this test simple if we can, otherwise the subprocess will need to import mock, mock logging, and then add more asserts.

short term, this seems elegant, but also if you feel strongly then happy to switch it around

Reply:
I'd be content with only testing that the process didn't crash (eg: returncode 0).

Maybe do something like
```
"Finished span not connected to a trace, adding to trace." in stderr
```
 instead of `stderr.startswith()`?

That way it's at least less susceptible to something else logging before this hits.